### DISCUSSION TITLE
TypeScript + Vuex4 + Quasar V2 beta

### DISCUSSION THREAD

**Discussion by armenr** [quasarframework/quasar#8715] [General - Components / Directives / etc]
So, looking around the web, I've found articles like this one: https://dev.to/3vilarthas/vuex-typescript-m4j

My question is simple: With the provided `import { store } from 'quasar/wrappers'`, do I need to go through all of the steps outlined in the article, as far as augmentation, and Omits, etc...or is this all provided through the store wrapper Quasar provides?

For the record, I've definitely read through the provided documentation at https://next.quasar.dev/ .

e.g - 

```
export type Store = Omit<
  VuexStore<State>,
  'getters' | 'commit' | 'dispatch'
> & {
  commit<K extends keyof Mutations, P extends Parameters<Mutations[K]>[1]>(
    key: K,
    payload: P,
    options?: CommitOptions
  ): ReturnType<Mutations[K]>
} & {
  dispatch<K extends keyof Actions>(
    key: K,
    payload: Parameters<Actions[K]>[1],
    options?: DispatchOptions
  ): ReturnType<Actions[K]>
} & {
  getters: {
    [K in keyof Getters]: ReturnType<Getters[K]>
  }
}
```

**Comment by aarondreese** [comment]
I have the same question.  As Q2 is already extending the Vuex store itself to incorporate all of the store modules and their associated mutations, actions and getters, where and how should we extend the store again to ensure enumeration type-safety happens correctly.  Should I do it in each module so that they are already in the relevant &lt;Tree&gt; when Q2 comes to merge them, Should I modify the Q2 store builder to get the enumerations from the modules, or should I be writing a separate wrapper around the Q2 store to explicitly do the work.  
Is type-safe enumeration on the development path for either Q2 or Vx4, or do we need to wait for Vx5
Setting up the enumerators is a PITA but if it has to be done....I'm sick of es-lint throwing errors every time I use Vx functionality.
